src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -0,0 +1,40 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.crypto.Keccak256;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+class PegoutSignedCacheImpl implements PegoutSignedCache {
+
+  private final Map<Keccak256, Long> cache = new ConcurrentHashMap<>();
+  private final Duration ttl;
+
+  PegoutSignedCacheImpl(Duration ttl) {
+    this.ttl = ttl;
+  }
+
+  @Override
+  public boolean hasAlreadyBeenSigned(Keccak256 pegoutCreationRskTxHash) {
+    Long currentTimestamp = System.currentTimeMillis();
+
+    return Optional.ofNullable(pegoutCreationRskTxHash)
+        .map(cache::get)
+        .map(timestamp -> isValidTimestamp(currentTimestamp, timestamp))
+        .orElse(false);
+  }
+
+  @Override
+  public void put(Keccak256 pegoutCreationRskTxHash) {
+    Long currentTimestamp = System.currentTimeMillis();
+
+    Optional.ofNullable(pegoutCreationRskTxHash)
+        .ifPresent(rskTxHash -> cache.put(rskTxHash, currentTimestamp));
+  }
+
+  private boolean isValidTimestamp(Long currentTimestamp, Long timestamp) {
+    return currentTimestamp != null && timestamp != null &&
+        currentTimestamp - timestamp <= ttl.toMillis();
```

Comment:
i'd convert this line into a variable, something like 
`timestampIsNotExpired = currentTimestamp - timestamp <= ttl.toMillis()`

so is easier to understand

Reply:
I feel like this is simple to understand. Also the null checks have to be done before evaluating the expression either way, so extracting it into a variable wouldn't help that much. Wdyt?

Reply:
Consider the case when a record has expired but the cleanup scheduler has not kicked in yet. If we call the method to check if it is present in the cache we would expect to return false.